src/logic/addressBook/store/actions/saveAndUpdateAddressBook.js

```diff
@@ -1,14 +1,20 @@
 // @flow
 import type { Dispatch as ReduxDispatch } from 'redux'
-import { type GlobalState } from '~/store/index'
+import { type GlobalState } from '~/safeStore'
 import { saveAddressBook } from '~/logic/addressBook/utils'
-import { updateAddressBook } from '~/logic/addressBook/store/actions/updateAddressBook'
+import { updateAddressBookEntry } from '~/logic/addressBook/store/actions/updateAddressBookEntry'
 import type { AddressBook } from '~/logic/addressBook/model/addressBook'
+import { get3Box } from '~/utils/storage'
 
 const saveAndUpdateAddressBook = (addressBook: AddressBook) => async (dispatch: ReduxDispatch<GlobalState>) => {
   try {
-    dispatch(updateAddressBook(addressBook))
-    await saveAddressBook(addressBook)
+    dispatch(updateAddressBookEntry(addressBook))
+    try {
+      await get3Box()
```

Comment:
why do we get 3box there and then not use it?

Reply:
It's just a promise that needs to be resolved, we can use or not the returned value.

In the current implementation, we only need to wait for the authentication/authorization to succeed before using 3box functions.